include/hip/hcc_detail/hip_vector_types.h

```diff
@@ -829,6 +830,26 @@ __MAKE_VECTOR_TYPE__(double, double);
         type r{x, y, z, w}; \
         return r; \
     }
+#else
+ #define DECLOP_MAKE_ONE_COMPONENT(comp, type) \
+     static inline __device__ __host__ \
+     type make_##type(comp x) { type r; r.x =x; return r; }
+
+ #define DECLOP_MAKE_TWO_COMPONENT(comp, type) \
+     static inline __device__ __host__ \
+     type make_##type(comp x, comp y) { type r; r.x=x; r.y=y; return r; }
+
+ #define DECLOP_MAKE_THREE_COMPONENT(comp, type) \
+     static inline __device__ __host__ \
+     type make_##type(comp x, comp y, comp z) { type r; r.x=x; r.y=y; r.z=z; return r; }
+
+ #define DECLOP_MAKE_FOUR_COMPONENT(comp, type) \
+     static inline __device__ __host__ \
+     type make_##type(comp x, comp y, comp z, comp w) { \
+         type r; r.x=x; r.y=y; r.z=z; r.w=w; \
```

Comment:
Instead of creating another set of macros, would it work if changing the initializer into:
`type r = {x,y,z,w};`

Reply:
I think clang compiler has some know issue when "<type> var = {}"  used in the code hence I was hitting compilation error. Skipped this as of now.......I see some where it is mentioned that in latest clang compiler they have addressed the issue. Will confirm and revisit the change in next PR.

Reply:
>I think clang compiler has some know issue when " var = {}" 

I'm unaware of such issue in clang, could you please confirm?

Reply:
Please refer below links:
1. https://jira.mongodb.org/browse/SERVER-26651
2. http://lists.llvm.org/pipermail/cfe-dev/2013-January/027385.html